akd_local_auditor/src/storage/s3/test.rs

```diff
@@ -60,14 +60,40 @@ pub async fn maybe_flush_storage(shared_config: &SdkConfig, bucket_name: &str) -
 
             // From: https://docs.aws.amazon.com/sdk-for-rust/latest/dg/rust_s3_code_examples.html
             let objects = client.list_objects_v2().bucket(bucket_name).send().await?;
-            // let mut delete_objects: Vec<ObjectIdentifier> = vec![];
+
+            let object_versions = client
+                .list_object_versions()
+                .bucket(bucket_name)
+                .max_keys(1000)
+                .send()
+                .await?;
+            let map = object_versions
+                .versions()
+                .unwrap()
+                .iter()
+                .map(|obj_ver| {
+                    (
+                        obj_ver.key().unwrap().to_string(),
+                        obj_ver.version_id().unwrap().to_string(),
+                    )
+                })
+                .collect::<std::collections::HashMap<String, String>>();
+            let get_version = |key: &str| {
+                let result = map.get(key).cloned();
```

Comment:
I think you can call `map_or_else` and panic if you hit the else. I think this may be sightly more readable and (potentially) microscopically more performant since you won't always call `clone`.

Reply:
No matter what, here we need to clone, since we want an `Option<String>` not an `Option<&String>` which wouldn't support downstream comparisons on equality I think.